src/coreclr/jit/ifconversion.cpp

```diff
@@ -769,6 +760,25 @@ bool OptIfConversionDsc::optIfConvert()
         }
     }
 
+    if (!m_comp->compStressCompile(Compiler::STRESS_IF_CONVERSION_INNER_LOOPS, 25))
+    {
+        // Don't optimise the block if it is inside a loop
+        // When inside a loop, branches are quicker than selects.
+        // Detect via the block weight as that will be high when inside a loop.
+        if (m_startBlock->getBBWeight(m_comp) > BB_UNITY_WEIGHT)
+        {
+            JITDUMP("Skipping if-conversion inside loop (via weight)\n");
+            return false;
+        }
```

Comment:
Would it be better to leave this "cheap" weight check first in the function, and only insert the "expensive" check here, after the long sequence of potentially expensive (appearing) checks above?

Reply:
I actually tried this locally, but I did not see any impact from moving the cheap check.

Reply:
It seems like that would do the `optReachable` check before identifying the candidate which is what I wanted to avoid, as the flow-graph walk is expensive. Also, it is using `m_finalBlock` which is found as part of identifying the candidate. I moved the other check to simplify the code (as it does not have noticeable impact).

Of course if the flow graph walk was prohibitively expensive we could just compute reachability sets in this phase, but since the TP impact of this approach didn't turn out too bad I think we can leave it like this.

Reply:
ah, didn't spot the use of final block, ok, agreed, leave as is.